Core/include/Acts/Material/ISurfaceMaterial.hpp

```diff
@@ -130,10 +130,14 @@ class ISurfaceMaterial {
 
 inline double ISurfaceMaterial::factor(NavigationDirection pDir,
                                        MaterialUpdateStage mStage) const {
-  if (mStage == Acts::fullUpdate) {
+  if (mStage == Acts::MaterialUpdateStage::FullUpdate) {
     return 1.;
   }
-  return (pDir * mStage > 0 ? m_splitFactor : 1. - m_splitFactor);
+  return (
+      pDir * static_cast<std::underlying_type_t<MaterialUpdateStage>>(mStage) >
+              0
+          ? m_splitFactor
+          : 1. - m_splitFactor);
```

Comment:
This is the only thing I don't really like, it smells a little unmaintainable to me. In my opinion it would be clearer to write:

```cpp
if (mStage == Acts::MaterialUpdateStage::PreUpdate) {
    return pDir < 0 ? m_splitFactor : 1. - m_splitFactor;
} else if (mStage == Acts::MaterialUpdateStage::PostUpdate) {
    return pDir > 0 ? m_splitFactor : 1. - m_splitFactor;
} else {
    return 1.;
}
```

Reply:
Fair enough, here it's not actually more concise with the multiplication. I'll change this.

Reply:
Updated, and also added a test for this behavior.